query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -75,9 +74,11 @@ private Object extractValue(Object wrappedValue) {
    public Void call() throws Exception {
       preIndex();
       KeyValueFilter filter = getFilter();
-      try (CloseableIterator<CacheEntry<Object, String>> iterator = entryRetriever.retrieveEntries(filter, null, Util.asSet(Flag.CACHE_MODE_LOCAL), null)) {
+      try (Stream<CacheEntry<Object, Object>> stream = cache.getAdvancedCache().withFlags(Flag.CACHE_MODE_LOCAL)
+              .cacheEntrySet().stream()) {
```

Comment:
We sure could.  I just didn't want to go changing the performance of this right away without being able to test if it was better or not.